File: src/ts/magma/app/compile/define/Whitespace.ts
import { Definition } from "../../../../magma/app/compile/define/Definition";
import { Option } from "../../../../magma/api/option/Option";
import { None } from "../../../../magma/api/option/None";
export class Whitespace {
	generate(): string {
		return ""/*unknown*/;
	}
	asDefinition(): Option<Definition> {
		return new None<Definition>()/*unknown*/;
	}
	is(type: string): boolean {
		return "whitespace".equals(type)/*unknown*/;
	}
}

File: src/java/magma/app/compile/define/Whitespace.java
package magma.app.compile.define;

import magma.api.option.None;
import magma.api.option.Option;
import magma.app.compile.node.Node;

public record Whitespace() implements Parameter, Node {
    public String generate() {
        return "";
    }

    public Option<Definition> asDefinition() {
        return new None<Definition>();
    }

    @Override
    public boolean is(String type) {
        return "whitespace".equals(type);
    }
}
